src/adapters/curve/products/pool/curvePoolAdapter.ts

```diff
@@ -187,4 +248,58 @@ export class CurvePoolAdapter
 
     return poolMetadata
   }
+
+  private async getPoolData(i: number): Promise<
+    | {
+        protocolToken: Erc20Metadata
+        underlyingTokens: Erc20Metadata[]
+        poolAddress: string
+      }
+    | undefined
+  > {
+    const metaRegistryContract = MetaRegistry__factory.connect(
+      CURVE_META_REGISTRY_CONTRACT,
+      this.provider,
+    )
+    const poolAddress = await metaRegistryContract.pool_list(i)
+    const lpToken = await metaRegistryContract['get_lp_token(address)'](
+      poolAddress,
+    )
+    const lpTokenContract = Erc20__factory.connect(lpToken, this.provider)
+
+    const underlyingCoins = (
+      await metaRegistryContract['get_underlying_coins(address)'](poolAddress)
+    ).filter((address) => address !== ZERO_ADDRESS)
```

Comment:
I've noticed in their metadata that they use 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee for ETH. Does that need checking here?

Reply:
I believe they use Zero_address to represent empty slots in an array because vyper language does not have dynamic arrays